File: activitynext/types/MessageNotificationDTO.ts
export interface MessageNotificationDTO {
  id: number;
  type: NotificationType | keyof typeof NotificationType;
  isRead: boolean;
  createdAt: string;
  readAt?: string | null;
  messageId?: number | null;
  conversationId?: number | null;
  senderId?: number | null;    
  senderName?: string | null;
  groupName?: string | null;
  groupImageUrl?: string;
  reactionEmoji?: string | null;
  messagePreview?: string | null;
  messageCount?: number | null;
  isTemporary?: boolean;
  isConversationRejected?: boolean;
  isReactionUpdate?: boolean;
  senderProfileImageUrl?: string | null;
}

export enum NotificationType {
  NewMessage = 1,
  MessageRequest = 2,
  MessageRequestApproved = 3,
  MessageReaction = 4,
  GroupRequest = 5,
  GroupRequestApproved = 6,
}

File: AFBack/DTOs/MessageNotificationDTO.cs
using System.Text.Json.Serialization;

namespace AFBack.Models;

public class MessageNotificationDTO
{
    public int Id { get; set; }
    
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public NotificationType Type { get; set; }
    public bool IsRead { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? ReadAt { get; set; }

    public int? MessageId { get; set; }
    public int? ConversationId { get; set; }
    public int? SenderId { get; set; }
    public string? SenderName { get; set; }
    
    public string? SenderProfileImageUrl { get; set; }
    public string? GroupName { get; set; }
    
    public string? GroupImageUrl { get; set; }
    
    [JsonPropertyName("reactionEmoji")]
    public string? ReactionEmoji { get; set; }
    
    public bool IsReactionUpdate { get; set; } = false;

    public string? MessagePreview { get; set; } // For "Ola: Hei!"-type visninger
    public int? MessageCount { get; set; }
    
    public bool? IsConversationRejected { get; set; }
    // Visning av hendelser
    public List<string>? EventSummaries { get; set; }
}